tests/integration/helm/upgrade/util.go

```diff
@@ -227,14 +226,94 @@ func performRevisionUpgradeFunc(previousVersion string) func(framework.TestConte
 		sanitycheck.RunTrafficTestClientServer(t, oldClient, newServer)
 	}
 }
+
+// performRevisionTagsUpgradeFunc returns the provided function necessary to run inside of a integration test
+// for upgrade capability with stable label revision upgrades
+func performRevisionTagsUpgradeFunc(previousVersion string) func(framework.TestContext) {
+	return func(t framework.TestContext) {
+		cs := t.Clusters().Default().(*kubecluster.Cluster)
+		h := helm.New(cs.Filename(), filepath.Join(previousChartPath, previousVersion))
+
+		//t.ConditionalCleanup(func() {
+		//	err := deleteIstioRevision(h, revisionChartSuffix)
+		//	if err != nil {
+		//		t.Fatalf("could not delete istio: %v", err)
+		//	}
+		//	err = deleteIstio(cs, h)
+		//	if err != nil {
+		//		t.Fatalf("could not delete istio: %v", err)
+		//	}
+		//})
+
+		// install 1.10.0 charts with revision set to "1-10-0"
+		previousRevision := strings.ReplaceAll(previousVersion, ".", "-")
+		overrideValuesFile := getValuesOverrides(t, gcrHub, previousVersion, previousRevision)
+		helmtest.InstallIstioWithRevision(t, cs, h, tarGzSuffix, "-"+previousRevision, overrideValuesFile, true)
+		helmtest.VerifyInstallation(t, cs, false)
+		helmtest.SetRevisionTag(t, h, tarGzSuffix, previousRevision, prodTag)
+		helmtest.VerifyMutatingWebhookConfigurations(t, cs, []string{"istio-revision-tag-prod", "istio-sidecar-injector-1-10-0"})
+
+		// setup istio.io/rev tag to point to previous version (e.g. istio.io/rev=1-10-0)
+		oldNs, oldClient, oldServer := sanitycheck.SetupTrafficTest(t, t, previousRevision)
+		sanitycheck.RunTrafficTestClientServer(t, oldClient, oldServer)
+
+		// now upgrade istio to the latest version found in this branch
+		// use the command line or environmental vars from the user to set
+		// the hub/tag
+		s, err := image.SettingsFromCommandLine()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// install the latest charts with revision set to "latest"
+		overrideValuesFile = getValuesOverrides(t, s.Hub, s.Tag, latestRevisionTag)
+		helmtest.InstallIstioWithRevision(t, cs, h, tarGzSuffix, "-"+latestRevisionTag, overrideValuesFile, false)
+		helmtest.VerifyInstallation(t, cs, false)
+		helmtest.SetRevisionTag(t, h, tarGzSuffix, latestRevisionTag, canaryTag)
+		helmtest.VerifyMutatingWebhookConfigurations(t, cs, []string{"istio-revision-tag-prod", "istio-sidecar-injector-1-10-0","istio-revision-tag-canary", "istio-sidecar-injector-latest"})
+
+		// setup istio.io/rev=latest
+		_, newClient, newServer := sanitycheck.SetupTrafficTest(t, t, latestRevisionTag)
+		sanitycheck.RunTrafficTestClientServer(t, newClient, newServer)
+
+		// now check that we are compatible with N-1 proxy with N proxy
+		sanitycheck.RunTrafficTestClientServer(t, oldClient, newServer)
+
+		// set the revision tag for prod to point to the latest istiod
+		helmtest.SetRevisionTag(t, h, tarGzSuffix, latestRevisionTag, prodTag)
+
+		err = oldNs.RemoveLabel(istioRevLabel)
```

Comment:
nit: should be able to just `SetLabel` without having to remove first?

Reply:
fixed